roachtest: remove perf artifacts dir when reusing a cluster

ajwerner opened the pull request:
Before this PR we'd leave the old perf artifacts dir in place. This had the
unfortunate consequence of polluting some roachperf data with results from
different tests.

Relates to #39634 but doesn't close it because we still need to go back and
clean things up.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/39647)
<!-- Reviewable:end -->

ajwerner commented:
cc @andreimatei if you're interested

Commit:
roachtest: remove perf artifacts dir when reusing a cluster

Before this PR we'd leave the old perf artifacts dir in place. This had the
unfortunate consequence of polluting some roachperf data with results from
different tests.

Relates to #39634 but doesn't close it because we still need to go back and
clean things up.

Release note: None

ajwerner commented:
TFTR!

bors r=nvanbenschoten
